go-controller/pkg/ovn/policy.go

```diff
@@ -924,6 +924,9 @@ func (oc *Controller) destroyNetworkPolicy(np *networkPolicy, nsInfo *namespaceI
 // ingress/egress address set
 func (oc *Controller) handlePeerPodSelectorAddUpdate(gp *gressPolicy, obj interface{}) {
 	pod := obj.(*kapi.Pod)
+	if pod.Spec.NodeName == "" {
```

Comment:
nice catch @girishmg Is there any scenario where the pod is scheduled but does not have an IP? Would we also want to skip in that case?

Reply:
@trozet I think there is a possibility of that happening. Say, we failed to add a logical switch port for that Pod and then when policy's peer pod handler gets called for that pod, we will blindly update the addressSet.

however, i think we have an issue with the policy's callback handler since it doesn't have retry logic on failures.